receiver/mongodbatlasreceiver/internal/mongodb_atlas_client.go

```diff
@@ -130,22 +130,33 @@ type MongoDBAtlasClient struct {
 
 // NewMongoDBAtlasClient creates a new MongoDB Atlas client wrapper
 func NewMongoDBAtlasClient(
+	baseURL string,
 	publicKey string,
 	privateKey string,
 	backoffConfig configretry.BackOffConfig,
 	log *zap.Logger,
-) *MongoDBAtlasClient {
+) (*MongoDBAtlasClient, error) {
 	defaultTransporter := http.DefaultTransport.(*http.Transport)
 	t := digest.NewTransportWithHTTPTransport(publicKey, privateKey, defaultTransporter)
 	roundTripper := newClientRoundTripper(t, log, backoffConfig)
 	tc := &http.Client{Transport: roundTripper}
-	client := mongodbatlas.NewClient(tc)
+
+	if baseURL == "" {
+		baseURL = mongodbatlas.CloudURL
+	}
+
+	client, err := mongodbatlas.New(tc, mongodbatlas.SetBaseURL(baseURL))
+	if err != nil {
+		log.Error("Failed to create client", zap.Error(err))
```

Comment:
```suggestion
```

If we return an error here we generally are not going to start the component and the returned value is going to be logged anyway. I'd of removing this error log line